### PR TITLE
move several methods between the compute client and the k8s client.

### PR DESCRIPTION
**Pull request by steeling**
With our current refactoring the k8s client will become a simple accessor with little/no business logic,
whereas the compute.Client will have the business logic for translating osm objects to/from k8s objects.
This also brings us closer to removing the kubecontroller from the meshcatalog

Signed-off-by: Sean Teeling <seanteeling@microsoft.com>

### CODE REVIEW

**Comment by steeling** [comment]
@keithmattix  can you take a look at this with respect to me removing the `GetServiceFromHostname` and `GetSubdomainFromHostname` which I believe was an eroeneous check due to splits not working outside of the namespace, nor with statefulsets

**Comment by keithmattix** [comment]
@steeling Will give a more thorough review later, but you're correct; good catch! Statefulsets aren't supported with TrafficSplits, so there's no need to add that extra logic

File: pkg/k8s/osm_injector_test.go [hunk 1]
```diff
@@ -1,102 +0,0 @@
...
-	tassert "github.com/stretchr/testify/assert"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/kubernetes/fake"
-
-	"github.com/openservicemesh/osm/pkg/constants"
-)
-
-func TestGetOSMInjectorPods(t *testing.T) {
```
Comment (keithmattix) [inline pkg/k8s/osm_injector_test.go:15]: This test doesn't appear to be replaced

**Review by keithmattix — COMMENTED** [review]
Looks good generally, but I see several deleted test files that don't appear to be replaced or accounted for in added tests. Holding off on an approval pending an an explanation

File: pkg/k8s/osm_injector_test.go (hunk 1 above)
Comment (steeling) [reply pkg/k8s/osm_injector_test.go:15]: The method in the corresponding file was deleted

**Comment by steeling** [comment]
> Looks good generally, but I see several deleted test files that don't appear to be replaced or accounted for in added tests. Holding off on an approval pending an an explanation

I believe those are all due to removing the original method

**Review by keithmattix — APPROVED** [review]

**Review by shashankram — COMMENTED** [review]
Needs a rebase after my recent lds changes.

**Comment by steeling** [comment]
> Needs a rebase after my recent lds changes.

done

File: pkg/cli/verifier/envoy_config.go [hunk 2]
```diff
@@ -461,7 +460,7 @@ func (v *EnvoyConfigVerifier) verifyDestination() Result {
 	return result
 }
 
-func (v *EnvoyConfigVerifier) getDstMeshServicesForSvcPod(svc corev1.Service, podName string) ([]service.MeshService, error) {
+func (v *EnvoyConfigVerifier) getDstMeshServicesForSvcPod(svc corev1.Service) ([]service.MeshService, error) {
```
Comment (shashankram) [inline pkg/cli/verifier/envoy_config.go:463]: nit: rename function to `getDstMeshServicesForK8sSvc`

File: pkg/compute/kube/fake/fake.go [hunk 3]
```diff
@@ -95,3 +96,32 @@ func (f fakeClient) GetResolvableEndpointsForService(svc service.MeshService) []
...
 	return endpoints
 }
+
+func (f fakeClient) IsMetricsEnabled(*envoy.Proxy) (bool, error) {
+	return true, nil
+}
+
+// GetHostnamesForService returns the hostnames over which the service is accessible
+func (f fakeClient) GetHostnamesForService(svc service.MeshService, localNamespace bool) []string {
+	var hostnames []string
```
Comment (shashankram) [inline pkg/compute/kube/fake/fake.go:106]: seems like the fake has the same implementation as the actual one. Can this be consolidated?

**Review by shashankram — APPROVED** [review]
Could you confirm the test coverage hasn't been significantly reduced after this refactor. Would be good to have codecov bot auto-detect this, which currently isn't configured to flag code coverage drops.

**Comment by keithmattix** [comment]
@shashankram CodeCov code coverage computation has been VERY unreliable in my experience. It's just anecdotal, but I'm hesitant to experience that again 😅

**Comment by shashankram** [comment]
> @shashankram CodeCov code coverage computation has been VERY unreliable in my experience. It's just anecdotal, but I'm hesitant to experience that again 😅

Which computation are you referring to? There's a GH action that is meant to reliably detect drop in coverage. The codecov bot that comments on PRs is known to be ambiguous.

**Comment by keithmattix** [comment]
Ah, I was referring to the codecov bot that comments on PRs. I'm open to a more consistent GitHub action 👍🏾

File: pkg/compute/kube/fake/fake.go (hunk 3 above)
Comment (steeling) [reply pkg/compute/kube/fake/fake.go]: done

File: pkg/cli/verifier/envoy_config.go (hunk 2 above)
Comment (steeling) [reply pkg/cli/verifier/envoy_config.go]: done

**Comment by steeling** [comment]
> Could you confirm the test coverage hasn't been significantly reduced after this refactor. Would be good to have codecov bot auto-detect this, which currently isn't configured to flag code coverage drops.

Test coverage percent has dropped in package k8s, but mostly due to deleting a few large methods that had 100% coverage (no longer needed), and slightly increasing the mocke kube client which has a very low percentage.

The overall lines of code tested looks to be more or less unchanged

**Pull request merged**